### PR TITLE
improvements in invite class & docs enhancements

### PR DESCRIPTION
**Pull request by Commandserver**
- fixed a bug where `invite.guild_id` is not set, because the json was wrong parsed
- added some attributes to `invite`
- docs

### CODE REVIEW

**Comment by Commandserver** [comment]
I've tested it both btw, `dpp::invite::uses` and `dpp::invite::expires_at`

File: src/dpp/cluster/invite.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ void cluster::invite_delete(const std::string &invitecode, command_completion_ev
 
 
 void cluster::invite_get(const std::string &invitecode, command_completion_event_t callback) {
-	this->post_rest(API_PATH "/invites", dpp::url_encode(invitecode) + "?with_counts=true", "", m_get, "", [callback](json &j, const http_request_completion_t& http) {
+	this->post_rest(API_PATH "/invites", dpp::url_encode(invitecode) + "?with_counts=true&with_expiration=true", "", m_get, "", [callback](json &j, const http_request_completion_t& http) {
```
Comment (Axyte) [inline src/dpp/cluster/invite.cpp:53]: This should be optional if possible.

File: include/dpp/invite.h [hunk 2]
```diff
@@ -66,6 +69,11 @@ class DPP_EXPORT invite {
 	/** True if this invite should not replace or "attach to" similar invites
 	 */
 	bool unique;
+	/** How many times this invite has been used
+	 *
+	 * @note Only set when using cluster::channel_invites_get
+	 */
+	uint32_t uses;
```
Comment (Axyte) [inline include/dpp/invite.h:76]: Set to zero in constructor alongside other added fields in the PR?

File: src/dpp/cluster/invite.cpp (hunk 1 above)
Comment (Axyte) [reply src/dpp/cluster/invite.cpp:53]: To provide choice to the user, of course.

**Pull request merged**